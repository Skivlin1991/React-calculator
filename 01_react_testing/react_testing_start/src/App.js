import './App.css';
import Counter from './components/Counter';
import Countries from './components/Countries'

const App = () => {
  return (
    <div className="App">
      <Counter />
      <Countries />
    </div>
  );
}

export default App;
