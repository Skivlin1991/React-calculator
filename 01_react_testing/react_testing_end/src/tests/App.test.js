import React from "react";
import { render } from '@testing-library/react';
import App from '../App';

describe('App', () => {
  it('should render react link', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot()
  });
})

