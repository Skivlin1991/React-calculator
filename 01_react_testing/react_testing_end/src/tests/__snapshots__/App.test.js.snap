// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1
      data-testid="counter"
    >
      0
    </h1>
    <button
      data-testid="button-up"
      id="button-up"
    >
       Up
    </button>
    <button
      data-testid="button-down"
      id="button-down"
    >
      Down
    </button>
    <div>
      <h2>
        Country Container
      </h2>
      <select
        data-testid="country-selector"
        id="country-selector"
      >
        <option
          disabled=""
          selected=""
          value="default"
        >
          Choose a country...
        </option>
      </select>
      <button
        data-testid="favourite-button"
        id="favourite-button"
      >
        Add to Favourites
      </button>
    </div>
  </div>
</DocumentFragment>
`;
